### DISCUSSION TITLE
How to force component to re-render view?

### DISCUSSION THREAD

**Discussion by gtamas** [Relm4/Relm4#609] [General]
I have a SimpleComponent which renders a form. Initially, one of the widgets (gtk::DropDown) is empty. But when the user clicks a button, the options become available, so the view should re-render and add options to the dropdown.

Is there an API to re-render the view? Or should I add the options to the dropdown widget manually and view will be updated?

What's the best way to do this?

**Comment by AaronErhardt** [comment]
You can send a message to the component, update the model and then the view is updated. During the view update, you can create and delete widgets using manual code: https://relm4.org/book/stable/component_macro/reference.html#child-widgets
